clippy_lints/src/assertions_on_constants.rs

```diff
@@ -63,12 +53,18 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for AssertChecks {
             then {
                 match inner.node {
                     LitKind::Bool(true) => {
-                        span_lint(cx, EXPLICIT_TRUE, e.span,
+                        span_lint(cx, ASSERTIONS_ON_CONSTANTS, e.span,
```

Comment:
The `span_lint_and_sugg` should work. The problem you had was in the `assert(true)` branch but you were still using `span_lint` there.

Reply:
@mikerite I check it here:
https://github.com/rust-lang/rust-clippy/pull/3582/files/7075015f311617710dcebfe6530bdca732287994#diff-fc99a0c4e187ac372040420fe4dcb24cR60
Test case:
https://github.com/rust-lang/rust-clippy/pull/3582/files/7075015f311617710dcebfe6530bdca732287994#diff-514c75729f8046fc08cc8bb536b2fd0aR13
Failed CI on this test:
https://travis-ci.com/rust-lang/rust-clippy/jobs/169168148#L1259